README.md

```diff
@@ -1,2 +1,432 @@
+
+```
+name: <string>
+namespace: <string>
+environment: <string>
+```
+
+### Cluster IP service template
+
+* Template file: `_cluster-ip-service.yaml`
+* Template name: `ffc-helm-library.cluster-ip-service`
+
+A K8s `Service` object of type `ClusterIP`.
+
+A basic usage of this object template would involve the creation of `templates/cluster-ip-service.yaml` in the parent Helm chart (e.g. `ffc-microservice`) containing:
+
+```
+{{- include "ffc-helm-library.cluster-ip-service" (list . "ffc-microservice.service") -}}
+{{- define "ffc-microservice.service" -}}
+# Microservice specific configuration in here
+{{- end -}}
+```
+
+#### Required values
+
+The following values need to be set in the parent chart's `values.yaml` in addition to the globally required values [listed above](#all-template-required-values):
+
+```
+container:
+  port: <integer>
+```
+
+### Container template
+
+* Template file: `_container.yaml`
+* Template name: `ffc-helm-library.container`
+
+A template for the container definition to be used within a K8s `Deployment` object.
+
+A basic usage of this object template would involve the creation of `templates/_container.yaml` in the parent Helm chart (e.g. `ffc-microservice`). Note the `_` in the name. This template is part of the `Deployment` object definition and will be used in conjunction the `_deployment.yaml` template ([see below](#deployment-template)). As a minimum, `templates/_container.yaml` would define environment variables and liveness/readiness probes, e.g.:
+
+```
+{{- define "ffc-microservice.container" -}}
+env: <list>
+livenessProbe: <map>
+readinessProbe: <map>
+{{- end -}}
+```
+
+The liveness and readiness probes could take advantage of the helper templates for [http GET probe](#http-get-probe) and [exec probe](#exec-probe) defined within the library chart and described below.
+
+#### Required values
+
+The following values need to be set in the parent chart's `values.yaml` in addition to the globally required values [listed above](#all-template-required-values):
+
+```
+image: <string>
+container:
+  imagePullPolicy: <string>
+  readOnlyRootFilesystem: <boolean>
+  allowPrivilegeEscalation: <boolean>
+  requestMemory: <string>
+  requestCpu: <string>
+  limitMemory: <string>
+  limitCpu: <string>
+```
+
+#### Optional values
+
+The following values can optionally be set in the parent chart's `values.yaml` to enable a command with arguments to run within the container:
+
+```
+container:
+  command: <list of strings>
+  args: <list of strings>
+```
+
+### Deployment template
+
+* Template file: `_deployment.yaml`
+* Template name: `ffc-helm-library.deployment`
+
+A K8s `Deployment` object.
+
+A basic usage of this object template would involve the creation of `templates/deployment.yaml` in the parent Helm chart (e.g. `ffc-microservice`) that includes the template defined in `_container.yaml` template:
+
+```
+{{- include "ffc-helm-library.deployment" (list . "ffc-microservice.deployment") -}}
```

Comment:
I couldn't get this to work as an `include` because of the way the container object is mapped to it.  I had to declare the deployment as a template.

Has this approach been confirmed to be working?

Reply:
It works with `helm template` when running against PRs of the demo service repos, but will double check with `helm update --dry-run` to make sure. My understanding was that `include` does exactly the same as `template`, but it allows you to pass the output to other functions. The docs suggest it's preferable to use `include` over `template` (assuming it works the same!).